embrace-android-delivery/src/test/kotlin/io/embrace/android/embracesdk/internal/delivery/storage/PayloadStorageServiceImplTest.kt

```diff
@@ -118,4 +120,41 @@ class PayloadStorageServiceImplTest {
         val msg = logger.internalErrorMessages.last().throwable?.message
         assertEquals("Pruned payload storage", msg)
     }
+
+    @Test
+    fun `getUndeliveredPayloads only returns incomplete ones with a different process identifier than the current`() {
+        listOf("old_pid" to 100_000L, PROCESS_ID to 200_000L).forEach { appInstance ->
+            listOf(true, false).forEach { complete ->
+                listOf(CRASH, SESSION, LOG, NETWORK).forEach { type ->
+                    val metadata = StoredTelemetryMetadata(
+                        timestamp = appInstance.second,
+                        uuid = UUID,
+                        processId = appInstance.first,
+                        envelopeType = type,
+                        complete = complete,
+                    )
+                    service.store(metadata) { stream ->
+                        stream.write("test".toByteArray())
+                    }
+                }
+            }
+        }
+
+        with(service.getUndeliveredPayloads()) {
+            assertEquals(4, size)
```

Comment:
would suggest one more assertion that checks one of each payload type was delivered

Reply:
Good call. Will add